libs/wingsdk/src/target-tf-gcp/function.ts

```diff
@@ -108,6 +116,49 @@ export class Function extends cloud.Function {
     );
   }
 
+  public addBucketPermission(
```

Comment:
Can we somehow make this generic? Like in the azure target [here](https://github.com/winglang/wing/blob/main/libs/wingsdk/src/target-tf-azure/function.ts#L198)